test/q-repeat.test.js

```diff
@@ -14,3 +15,56 @@ describe('map', () => {
     });
   });
 });
+
+describe('retry', () => {
+  it('repeats until success', () => {
+    let delay = 500;
+    let invocations = 0;
+
+    return Q.timeoutRetry(150, () => {
+      delay -= 100;
+      invocations += 1;
+      return Q(`answer ${invocations}`).delay(delay);
+    }).then((result) => {
+      expect(delay).to.eq(100);
+      expect(invocations).to.eq(4);
+      expect(result).to.eq('answer 4');
+    });
+  });
+
+  it('fails fast if actual error occurs', () => {
+    return Q.timeoutRetry(500, () => {
+      throw new Error('omg');
+    }).catch((err) => {
+      expect(err.message).to.eq('omg');
+    });
+  });
+
+  it('hits outer timeout', () => {
```

Comment:
The test isn't there because I'm trying to encourage the pattern.  It's to check behavior if it does end up embedded withing a much broader timeout context.

That is, in effect, how it will applied in lambdas.  But when the big outer timeout times out, it's to kill execution, not to keep chomping on something else.